src/dune_rules/merlin/merlin.ml

```diff
@@ -556,6 +613,21 @@ module Unprocessed = struct
       ~f:(pp_flags ctx ~expander t.config.libname)
   ;;
 
+  let add_lib_dirs sctx mode ~init libs =
```

Comment:
Can you give `init` a better (outside) name? It is used as a named argument to `List.fold_left` but in the context of `val add_lib_dirs : Super_context.t -> Mode.t -> init: -> ...`  is quite nondescript.

Reply:
I now make the union later in [b4641d9](https://github.com/ocaml/dune/pull/10535/commits/b4641d937e68d578e6417de90416a731656e1f29), and since `source_dir` was always empty I removed it altogether.